indico/web/client/js/react/components/DateRangePicker.jsx

```diff
@@ -0,0 +1,224 @@
+import {Param, Translate} from 'indico/react/i18n';
+import {DateRange} from 'indico/utils/date';
+import {formatDate} from 'indico/utils/date_format';
+import {fromISOLocalDate} from 'indico/utils/date_parser';
+
+import 'indico/custom_elements/ind_date_picker';
+
+const INVALID = '__invalid__';
+
+export default function DateRangePicker({
+  value,
+  label,
+  format,
+  rangeStartLabel,
+  rangeEndLabel,
+  rangeStartMin,
+  rangeStartMax,
+  rangeEndMin,
+  rangeEndMax,
+  readOnly,
+  disabled,
+  onChange,
+}) {
+  format ??= moment.localeData().longDateFormat('L');
+
+  function handleChange(ev) {
+    const picker = ev.currentTarget;
+    const [start, end] = picker.dateRange;
+    const rangeStartInput = picker.querySelector('[data-range-start]');
+    const rangeEndInput = picker.querySelector('[data-range-end]');
+
+    const startInvalid = rangeStartInput.value && !start;
+    const endInvalid = rangeEndInput.value && !end;
+
+    onChange({
+      // ISO local dates
+      startDate: startInvalid ? INVALID : start?.toLocaleDateString('en-ca') || null,
+      endDate: endInvalid ? INVALID : end?.toLocaleDateString('en-ca') || null,
+    });
+  }
+
+  return (
+    <ind-date-range-picker
+      range-start={fromISOLocalDate(value.startDate)?.toDateString() || ''}
+      range-end={fromISOLocalDate(value.endDate)?.toDateString() || ''}
+      range-start-min={rangeStartMin}
+      range-start-max={rangeStartMax}
+      range-end-min={rangeEndMin}
+      range-end-max={rangeEndMax}
+      onChange={handleChange}
+    >
+      <fieldset>
+        <legend>{label}</legend>
+        <label>
+          <span>{rangeStartLabel}:</span>
+          <input
+            type="text"
+            data-range-start
+            placeholder={format}
+            readOnly={readOnly}
+            disabled={disabled}
+          />
+        </label>
+        <span className="arrow" />
+        <label>
+          <span>{rangeEndLabel}:</span>
+          <input
+            type="text"
+            data-range-end
+            placeholder={format}
+            readOnly={readOnly}
+            disabled={disabled}
+          />
+        </label>
+        <button type="button" data-calendar-trigger>
+          <Translate as="span">Open a calendar</Translate>
+        </button>
+
+        <span className="date-format" data-format>
+          <Translate>
+            Date format: <Param name="format" value={format} />
+          </Translate>
+        </span>
+      </fieldset>
+
+      <DatePickerCalendar>
+        <div className="calendars">
+          <DatePickerCalendar.Grid includeMonthHeader />
+          <DatePickerCalendar.Grid includeMonthHeader />
+        </div>
+      </DatePickerCalendar>
+    </ind-date-range-picker>
+  );
+}
+
+DateRangePicker.propTypes = {
+  format: PropTypes.string,
+  onChange: PropTypes.func.isRequired,
+  readOnly: PropTypes.bool,
+  disabled: PropTypes.bool,
+  value: PropTypes.object,
```

Comment:
I'd use a more precise type here (helps with autocomplete :D)
```suggestion
  value: PropTypes.shape({startDate: PropTypes.string.isRequired, endDate: PropTypes.string.isRequired}),
```

Reply:
+1 on this, I remember when I was having weird problems while testing due to me passing a moment object instead of a string, and a proptypes warning would have probably helped in noticing this more quickly